card/mod/05_standard/set/abstract/attachment.rb

```diff
@@ -110,6 +110,13 @@ def upload_cache_card
   @upload_cache_card ||= Card["new_#{attachment_name}".to_sym ]
 end
 
+def cached_upload= value
+  @cached_upload = value
+end
+
+def cached_upload
+  @cached_upload
```

Comment:
this is the only thing I'm really seeing that needs to happen to get this PR merged.  seems like the val of @cached_upload would either be a File or true/false.  Hard to guess that it's an action_id.